test/unit/general/https-support-test.js

```diff
@@ -0,0 +1,451 @@
+                            './test/unit/examples/subscriptionResponses/simpleSubscriptionSuccess.json'));
+
+                iotAgentLib.clearAll(function() {
+                    request(optionsProvision, function(error, result, body) {
+                        done();
+                    });
+                });
+            });
+        });
+
+        afterEach(function(done) {
+            nock.cleanAll();
+            iotAgentLib.setNotificationHandler();
+            iotAgentLib.clearAll(function() {
+                iotAgentLib.deactivate(done);
+            });
+        });
+
+        it('should send the appropriate request to the Context Broker', function(done) {
+            iotAgentLib.getDevice('MicroLight1', 'smartGondor', '/gardens', function(error, device) {
+                iotAgentLib.subscribe(device, ['attr_name'], null, function(error) {
+                    should.not.exist(error);
+
+                    contextBrokerMock.done();
+
+                    done();
+                });
+            });
+        });
+    });
+
+    describe('When a new device is connected to the IoT Agent', function() {
+        beforeEach(function(done) {
+            nock.cleanAll();
+
+            iotAgentConfig.iotManager = {};
+
+            contextBrokerMock = nock('https://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', 'gardens')
+                .post('/NGSI9/registerContext', utils.readExampleFile(
+                    './test/unit/examples/contextAvailabilityRequests/registerIoTAgent1.json'))
+                .reply(200, utils.readExampleFile(
+                    './test/unit/examples/contextAvailabilityResponses/registerIoTAgent1Success.json'));
+
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', 'gardens')
+                .post('/v1/updateContext')
+                .reply(200, utils.readExampleFile(
+                    './test/unit/examples/contextResponses/createProvisionedDeviceSuccess.json'));
+
+            iotAgentLib.activate(iotAgentConfig, function(error) {
+                iotAgentLib.clearAll(done);
+            });
+        });
+
+        it('should register as ContextProvider using HTTPS', function(done) {
+            iotAgentLib.register(device1, function(error) {
+                    should.not.exist(error);
+                    contextBrokerMock.done();
+                    done();
+            });
+        });
+
+        afterEach(function(done) {
+          nock.cleanAll();
+          iotAgentLib.clearAll(function() {
+              // We need to remove the registrationId so that the library does not consider next operatios as updates.
+              delete device1.registrationId;
+              iotAgentLib.deactivate(done);
+          });
+        });
+
+    });
+});
+
+describe('HTTPS support tests NGSIv2', function() {
```

Comment:
This breaks what we have done to include NGSIv2 tests. As you can see they are all located within https://github.com/telefonicaid/iotagent-node-lib/tree/master/test/unit/ngsiv2